### DISCUSSION TITLE
GraphQL and `$CURRENT_USER` Integrations?

### DISCUSSION THREAD

**Discussion by wrynegade** [directus/directus#10455] [Feature Requests]
Two related questions:

The "query" operation for `/graphql/system` GraphQL api seems to work fine for `users_me`,
```graphql
# this works!
query { users_me { id } }
```
but I can't seem to find documentation on how to make a mutation on this collection. I've all kinds of combinations of putting `update` in front of and `item(s)` on the end of `users_me` with no success. **Are mutations on `users_me` supported?**

As a workaround, I was hoping to use the `$CURRENT_USER` dynamic variable but I haven't been able to make this work in any request to system or item collections. **Is `$CURRENT_USER` supported within the GraphQL API?**

**Comment by rijkvanzanten** [comment]
> Are mutations on users_me supported?

Not on `users_me` necessarily. Updates on users are done through `update_users`.

> Is $CURRENT_USER supported within the GraphQL API?

Yes, but not for the purpose you're thinking of here. $CURRENT_USER can't be used in place of a primary key when doing mutations. It's only used in the arguments (like filter).

**Comment by wrynegade** [comment]
It's very convenient to use the REST API on `/users/me`; is there any plan to include mutations on the `users_me` alias?

**Reply by rijkvanzanten** [reply]
There wasn't a plan until now. I'll open a new issue to make sure we document and align the PATCH /users/me endpoint 🙂

**Reply by rijkvanzanten** [reply]
Ref https://github.com/directus/directus/issues/10500

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
